crates/common/consensus/src/deneb/beacon_state.rs

```diff
@@ -620,4 +624,46 @@ impl BeaconState {
 
         Ok(())
     }
+
+    pub fn process_attester_slashing(
+        &mut self,
+        attester_slashing: AttesterSlashing,
+    ) -> anyhow::Result<()> {
+        let attestation_1 = &attester_slashing.attestation_1;
+        let attestation_2 = &attester_slashing.attestation_2;
+
+        // Ensure the two attestations are slashable
+        ensure!(
+            is_slashable_attestation_data(&attestation_1.data, &attestation_2.data),
+            "Attestations are not slashable"
+        );
+
+        // Validate both attestations
+        ensure!(
+            is_valid_indexed_attestation(self, attestation_1),
+            "First attestation is invalid"
+        );
+        ensure!(
+            is_valid_indexed_attestation(self, attestation_2),
+            "Second attestation is invalid"
+        );
+
+        let current_epoch = self.get_current_epoch();
+        let indices_1: HashSet<_> = attestation_1.attesting_indices.iter().cloned().collect();
+        let indices_2: HashSet<_> = attestation_2.attesting_indices.iter().cloned().collect();
+
+        let mut slashed_any = false;
+
+        // Find common attesting indices and process slashing
+        for &index in indices_1.intersection(&indices_2).sorted() {
+            if is_slashable_validator(self.validators[index as usize].clone(), current_epoch) {
+                let _ = self.slash_validator(index, None);
```

Comment:
```suggestion
                let _ = self.slash_validator(index, None);
```

we need to handle the error here no?